releasenotes/notes/fix-pipeline-with-join-node-5f23a426cd4d88d9.yaml

```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Fixes pipelines using join nodes to properly pass on additional key value pairs from nodes prior to the join node to nodes that come after the join node.
```

Comment:
Since this release note will be published, can we try to make it more explanatory?

- I would classify this as an `enhancement` (more than a `fix`)
- Can you add an example here?

Reply:
> I would classify this as an enhancement (more than a fix)
We can agree to disagree on this one 😅 but I'm happy to relabel as an enhancement

> Can you add an example here?
Is it all right if I just provide the example I have [here](https://github.com/deepset-ai/haystack/pull/7510#issue-2232787891)?

Reply:
Ah, ok... Can be considered a fix.

The example you reported can be ok, with a short explanation.

Reply:
Okay it's been added!

Reply:
very comprehensive. Thanks!